integration-libs/punchout/core/facade/punchout.service.ts

```diff
@@ -6,25 +6,107 @@
 
 import { inject, Injectable } from '@angular/core';
 
+import { Command, CommandService, RoutingService } from '@spartacus/core';
 import {
+  PUNCHOUT_ERROR_PAGE_URL,
   PunchoutFacade,
   PunchoutRequisition,
   PunchoutSession,
 } from '@spartacus/punchout/root';
-import { Observable } from 'rxjs';
+
+import {
+  catchError,
+  forkJoin,
+  map,
+  Observable,
+  of,
+  switchMap,
+  throwError,
+} from 'rxjs';
 import { PunchoutConnector } from '../connectors';
+import { PunchoutAuthService } from '../services';
 
 @Injectable()
 export class PunchoutService implements PunchoutFacade {
   protected punchoutConnector = inject(PunchoutConnector);
+  protected punchoutAuthService = inject(PunchoutAuthService);
+  protected commandService = inject(CommandService);
+  protected routingService = inject(RoutingService);
+
+  /**
+   * getPunchoutSession workflow:
+   * Get PunchoutSession from  occ api
+   * Logout silently
+   * Login silently
+   * Route to target page based on punchout session info
+   * Redirect to Punchout Error page if error occurs
+   */
+  protected getPunchoutSessionCommand: Command<
+    { sessionId: string },
+    PunchoutSession
+  > = this.commandService.create((payload) => {
+    if (!payload?.sessionId) {
+      this.displayErrorPage();
+      return throwError(() => new Error('Punchout Session Id missing'));
+    }
+    return this.punchoutConnector.getPunchoutSession(payload.sessionId).pipe(
+      map((punchoutSession) => {
+        if (
+          !punchoutSession?.token?.accessToken ||
+          !punchoutSession?.customerId
+        ) {
+          throw new Error('Punchout login info missing');
+        }
+        return punchoutSession;
+      }),
+      switchMap((punchoutSession) => {
+        return forkJoin({
+          punchoutSession: of(punchoutSession),
+          logout: this.punchoutAuthService.logout(),
+        });
+      }),
+      map(({ punchoutSession }) => {
+        if (punchoutSession?.token?.accessToken) {
```

Comment:
This condition has been already checked on line 54, however linting does not let me without it.
Other solution is to add ts-ignore